compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs

```diff
@@ -82,22 +80,32 @@ pub enum PredicateFilter {
     SelfAndAssociatedTypeBounds,
 }
 
+#[derive(Debug)]
+pub enum RegionInferReason<'a> {
```

Comment:
Beautiful :sob: :heart:

Reply:
I guess we don't actually use most of the reasons but anyway...

Reply:
Yea, but that's true for obligation causes, too :D maybe we should. I dislike those delay span bugs